crates/wdk-build/src/lib.rs

```diff
@@ -712,19 +725,29 @@ impl Config {
     ///
     /// The iterator considers both the [`ApiSubset`] and the [`Config`] to
     /// determine which headers to yield
-    pub fn headers(&self, api_subset: ApiSubset) -> impl Iterator<Item = String> {
-        match api_subset {
+    ///
+    /// # Errors
+    /// [`ConfigError`] - if the headers for the given [`ApiSubset`] could not
+    /// be determined
+    pub fn headers(
+        &self,
+        api_subset: ApiSubset,
+    ) -> Result<impl Iterator<Item = String>, ConfigError> {
+        let headers = match api_subset {
             ApiSubset::Base => self.base_headers(),
             ApiSubset::Wdf => self.wdf_headers(),
             ApiSubset::Gpio => self.gpio_headers(),
             ApiSubset::Hid => self.hid_headers(),
             ApiSubset::ParallelPorts => self.parallel_ports_headers(),
             ApiSubset::Spb => self.spb_headers(),
             ApiSubset::Storage => self.storage_headers(),
-            ApiSubset::Usb => self.usb_headers(),
-        }
-        .into_iter()
-        .map(str::to_string)
+            ApiSubset::Usb => return self.usb_headers().map(std::iter::IntoIterator::into_iter),
```

Comment:
its awkward to specialcase this one branch to early return. should make them all uniform (which is why I suggests just standardising on CoW: https://github.com/microsoft/windows-drivers-rs/pull/411#pullrequestreview-3022563060

Reply:
We had a discussion about it. Using a `Cow` meant the return type of functions `base_headers`, `wdf_headers`, `gpio_headers`, `hid_headers`, `spb_headers`, `storage_headers` and `usb_headers` will all have to change to `Vec<Cow<`_, str>>` and therefore every element we put in in a `Vec` in each of them will have to made into a `Cow`. Compared to that doing an early return in one function seemed less awkward. That is not so it is not ugly in itself, but it seemed less ugly in comparison.

Reply:
Yes, all of them except `usb_headers` return `Vec<&str>`